tests/test_repository_tool.py

```diff
@@ -1299,22 +1320,20 @@ def test_delegate(self):
     self.assertEqual(self.targets_object.get_delegated_rolenames(),
                      ['tuf'])
 
-    # Test for targets that do not exist under the targets directory.
-    self.targets_object.revoke(rolename)
-    self.targets_object.delegate(rolename, public_keys, paths, threshold,
-        terminating=False, list_of_targets=['non-existent.txt'],
-        path_hash_prefixes=path_hash_prefixes)
-    for delegation in self.targets_object.delegations:
-      self.assertFalse('non-existent.txt' in delegation.target_files)
-
     # Test for delegated paths that do not exist.
     # An exception should not be raised for non-existent delegated paths, since
     # these paths may not necessarily exist when the delegation is done,
     # and also because the delegated paths can be glob patterns.
-    self.targets_object.delegate(rolename, public_keys, ['non-existent.txt'],
+    self.targets_object.delegate(rolename, public_keys, ['non-existent'],
         threshold, terminating=False, list_of_targets=list_of_targets,
         path_hash_prefixes=path_hash_prefixes)
 
+    # Test for delegated targets that do not exist.
+    # An exception should not be raised for non-existent delegated paths,
+    # since at this point the file system should not be accessed yet
+    self.targets_object.delegate(rolename, public_keys, [], threshold,
+        terminating=False, list_of_targets=['non-existent.txt'],
+        path_hash_prefixes=path_hash_prefixes)
```

Comment:
I'm not sure this test still makes sense?

Reply:
I kept the 'non-existent' target checks in all related tests  just to make sure that the file system is not accessed at this point and no exception is raised.